### DISCUSSION TITLE
Tree 组件中将节点拖拽到一个同级展开节点的下方时，drop indicator 很难出现

### DISCUSSION THREAD

**Discussion by willwu9304** [ant-design/ant-design#46772] [General]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/tuo-dong-shi-li-antd-5-12-4-forked-zxk66c?file=%2Fpackage.json%3A10%2C22)

### Steps to reproduce

按照视频所示进行拖拽，我分别试了不展开和展开，发现不展开时比较容易出现 indicator。但一旦打开，就很难出现了。
如果拖拽点往后一点又比较容易出现了。

https://github.com/react-component/tree/assets/45704808/15fb0d3a-db9a-42b0-bb14-b3c56c7ab4da

本地用的其实是 rc-tree，也有这个问题。但那个仓库响应问题比较慢，所以搬运到这里

### What is expected?

当拖拽点比较靠前，且同级目录展开的情况下，仍然能够拖动到其底部

### What is actually happening?

当拖拽点比较靠前，且同级目录展开的情况下，无法拖动到其底部

| Environment | Info |
| --- | --- |
| antd | 5.12.4 |
| React | v18.0.0 |
| System | MacOS |
| Browser | Chrome |

**Comment by MadCcc** [comment]
和 allowDrop 的逻辑有关，可以先自行调试一下

**Reply by willwu9304** [reply]
@MadCcc allowDrop 其实没有限制同级节点之间进行拖拽排序。在它们都是关闭状态是可以正常拖动的，如果打开情况，拖拽点在节点中间部位也可以的。

**Reply by willwu9304** [reply]
我精简了一下 allowDrop 的逻辑。出问题的是在第二条规则上：非叶子节点不允许拖到其他非叶子节点内部。

我希望将 0-0 拖拽到 0-2 的下方。如果 dropNode 是 0-2 中的最后一个叶子节点，则不管 dropPosition 是 -1 0 1，都是将 0-0 拖拽到 0-2 内部了对吧。这不是我期望的。

在 0-2 打开情况下，我期望 dropNode 能够是 0-2（dropPosition 是 1） 或者 0-3（dropPosition 是 -1）。但似乎在 0-2 打开的情况下始终无法触发。奇怪的是拖动点在 0-0 的中间部位，又很容易是 0-2 或者  0-3 了

https://github.com/ant-design/ant-design/assets/45704808/c762d428-415f-459c-9260-084e6f6dcb7f

**Reply by willwu9304** [reply]
为啥第二次打印 dragOver、dragOverGapTop、dragOverGapBottom 三个值都是 false？
